### PR TITLE
Create spoke account#212

### PR DESCRIPTION
**Pull request by jperata**

### CODE REVIEW

File: lib/external/spokes.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        try {
+            const response = await get(options);
+
+        } catch (error) {
+            if (error.statusCode && error.statusCode !== 404) {
+                // different kind of error, we log and throw
+                LoggingHelper.error(Logger, `Error while verifying id: ${ error.message }`);
+                throw error;
+            }
+            // uuid don't exist
```
Comment (jkelvie) [inline lib/external/spokes.ts]: "don't" should be "does not" (or "doesn't")

File: lib/external/spokes.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    name: this._id,
+                },
+                path: "/",
+                pipeType: "string",
+                proxy: false,
+            },
+            json: true // Automatically parses the JSON string in the response
+        };
+
+        const response = await post(options);
```
Comment (jkelvie) [inline lib/external/spokes.ts]: I would just have this return directly, rather than breaking into two statements

File: lib/external/spokes.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        } catch (error) {
+            if (error.statusCode && error.statusCode !== 404) {
+                // different kind of error, we log and throw
+                LoggingHelper.error(Logger, `Error while verifying id: ${ error.message }`);
+                throw error;
+            }
+            // uuid don't exist
+            return true;
+        }
+        return false;
```
Comment (jkelvie) [inline lib/external/spokes.ts]: I didn't realize it was possible to just directly return a value for a promise - how does that work? Does typescript automatically wrap the boolean result in a promise?

File: lib/external/spokes.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        const options = {
+            uri: `https://api.bespoken.link/pipe/${ this._secretKey }`,
+            headers: {
+                "x-access-token": "4772616365-46696f72656c6c61",
+            },
+            body: {
+
+            },
+            json: true // Automatically parses the JSON string in the response
+        };
```
Comment (jkelvie) [inline lib/external/spokes.ts]: Blank line here, after the end of this inline object.

File: lib/external/spokes.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        } catch (error) {
+            if (error.statusCode && error.statusCode !== 404) {
+                // different kind of error, we log and throw
+                LoggingHelper.error(Logger, `Error while verifying id: ${ error.message }`);
+                throw error;
+            }
+            // uuid don't exist
+            return true;
+        }
```
Comment (jkelvie) [inline lib/external/spokes.ts]: Blank line after this closing bracket

File: lib/external/spokes.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                path: "/",
+                pipeType: "string",
+                proxy: false,
+            },
+            json: true // Automatically parses the JSON string in the response
+        };
+
+        const response = await post(options);
+        return response;
+
```
Comment (jkelvie) [inline lib/external/spokes.ts]: This blank line should be removed

File: test/external/source-name-generator-test.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    mockery.warnOnUnregistered(false);
+    mockery.warnOnReplace(false);
+    mockery.registerMock("request-promise-native", {
+        get: function () {
+            return `{
+                "id": "id",
+                "secretKey": "secretKey"
+            }`;
+        }
+    });
```
Comment (jkelvie) [inline test/external/source-name-generator-test.ts]: Blank line after this statement

File: lib/external/spokes.ts (hunk 3 above)
Comment (jperata) [inline lib/external/spokes.ts]: yes, having async wrap everything in a promise.

File: lib/external/spokes.ts (hunk 2 above)
Comment (jperata) [inline lib/external/spokes.ts]: will do

File: lib/external/spokes.ts (hunk 4 above)
Comment (jperata) [inline lib/external/spokes.ts]: done

File: test/external/source-name-generator-test.ts (hunk 7 above)
Comment (jperata) [inline test/external/source-name-generator-test.ts]: done

**Pull request merged**